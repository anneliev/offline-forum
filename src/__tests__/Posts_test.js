import React from 'react';
import { shallow, mount } from 'enzyme';
import Posts from '../components/Posts';
import * as api from '../api';

describe('can show and remove comment', () =>{
  const posts = [
    {
      id: "1",
      title: "Zacs post title",
      content: "Zacs post content",
      author: "Zac",
      date: "2018-01-25 10:56:00",
    }
  ];
  
  it('should show a post from postList', () => {
    const wrapper = shallow(<Posts currentPersona="Zac" />);
    wrapper.setState({ posts });
    wrapper.instance().renderPostList(posts);
    expect(wrapper.state().posts[0].content).toEqual("Zacs post content");
  });
  
  it('should call remove post function', () => {
    api.removePost = jest.fn();
    const wrapper = shallow(<Posts currentPersona="Zac" postId="1" />);
    wrapper.instance().removePost();
    expect(api.removePost).toHaveBeenCalled();
  });
});

describe('can set post in localStorage', () => {
  const jsonPost = JSON.stringify([
    {
      id: "1",
      title: "Steffes post title",
      content: "Steffes post content",
      author: "Steffes",
      date: "2018-01-25 11:37:00",
    }
  ]);
  
  it('Should fetch post from localStorage', ()=> {
    const wrapper = mount(<Posts currentPersona="Steffe" /> );
    expect(wrapper.state().posts[0].content).toEqual('Steffes post content');
  });

  beforeEach(() => {
    localStorage.setItem('posts', jsonPost);
  });
  
  afterEach(() => {
    localStorage.clear();
  });
});

it('should match snapshot', () => {
  const wrapper = mount(<Posts currentPersona="Steffe" /> );
  expect(wrapper.html()).toMatchSnapshot();
})

