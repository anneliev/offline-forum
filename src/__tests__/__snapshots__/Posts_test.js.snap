// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `"<div class=\\"flex flex-wrap mx-auto pt-8\\"><form class=\\"container mx-auto flex flex-col p-6\\"><label for=\\"title\\" class=\\"mb-2\\">Title</label><input type=\\"text\\" value=\\"\\" name=\\"title\\" id=\\"title\\" class=\\"shadow rounded border mb-2 p-2\\"><label for=\\"content\\" class=\\"mb-2\\">Content </label><textarea id=\\"content\\" name=\\"content\\" class=\\"shadow rounded border mb-2 p-2\\"></textarea><input type=\\"submit\\" value=\\"Create\\" class=\\"bg-indigo-dark hover:bg-indigo-darker text-white font-bold py-2 px-4 rounded-full\\"></form></div>"`;
